src/apps/Permissions/AppCard.js

```diff
@@ -1,100 +1,92 @@
-import React from 'react'
+import React, { useCallback } from 'react'
 import PropTypes from 'prop-types'
-import styled from 'styled-components'
-import { Text, Card, Badge, theme, unselectable } from '@aragon/ui'
+import { Card, Tag, textStyle, unselectable, GU } from '@aragon/ui'
 import { AppType } from '../../prop-types'
 import { shortenAddress } from '../../web3-utils'
 import AppIcon from '../../components/AppIcon/AppIcon'
 
-class AppCard extends React.PureComponent {
-  static propTypes = {
-    app: AppType.isRequired,
-    onOpen: PropTypes.func.isRequired,
-  }
+const AppCard = React.memo(function AppCard({ onOpen, app }) {
+  const {
+    name,
+    identifier,
+    isAragonOsInternalApp,
+    hasWebApp,
+    proxyAddress,
+  } = app
 
-  handleClick = () => {
-    this.props.onOpen(this.props.app.proxyAddress)
-  }
-  render() {
-    const { app } = this.props
-    const {
-      name,
-      identifier,
-      isAragonOsInternalApp,
-      hasWebApp,
-      proxyAddress,
-    } = app
+  const onClick = useCallback(() => {
+    onOpen(proxyAddress)
+  }, [onOpen, proxyAddress])
 
-    const instanceTitle = `Address: ${proxyAddress}`
-    const instanceLabel = isAragonOsInternalApp
-      ? 'System App'
-      : !hasWebApp
-      ? 'Background App'
-      : identifier || shortenAddress(proxyAddress)
+  const instanceTitle = `Address: ${proxyAddress}`
+  const instanceLabel = isAragonOsInternalApp
+    ? 'System App'
+    : !hasWebApp
+    ? 'Background App'
+    : identifier || shortenAddress(proxyAddress)
 
-    return (
-      <Main onClick={this.handleClick}>
-        <div css="margin-bottom: 5px">
-          <AppIcon app={app} size={28} />
-        </div>
-        <Name>
-          <span>{name || 'Unknown'}</span>
-        </Name>
-        <IdentifierWrapper>
-          <Identifier title={instanceTitle}>{instanceLabel}</Identifier>
-        </IdentifierWrapper>
-        <Action>
-          <Text weight="bold" color={theme.textSecondary}>
-            View
-          </Text>
-        </Action>
-      </Main>
-    )
-  }
-}
-
-const Main = styled(Card).attrs({ width: '100%', height: '180px' })`
-  ${unselectable};
-  position: relative;
-  overflow: hidden;
-  display: flex;
-  flex-direction: column;
-  align-items: center;
-  padding-top: 25px;
-  cursor: pointer;
-`
-
-const Name = styled.p`
-  display: flex;
-  justify-content: center;
-  max-width: calc(100% - 20px);
-  margin: 0 10px 10px;
-  span {
-    display: block;
-    white-space: nowrap;
-    overflow: hidden;
-    text-overflow: ellipsis;
-  }
-`
+  return (
+    <Card
```

Comment:
Maybe the title should go at this level no? The tooltip may appear if the cursor is set over the whole thing instead only the `Tag`?

Reply:
`title`, in theory, takes precedence over the content defined inside, so I would rather keep it this way so the rest of the card content stays accessible.